src/adapter/edges.rs

```diff
@@ -527,84 +531,244 @@ pub(super) fn resolve_variant_edge<'a, V: AsVertex<Vertex<'a>> + 'a>(
     }
 }
 
+fn compute_variant_discriminants<'a>(
+    outer_item: &'a Item,
+    item_index: &'a HashMap<Id, Item>,
+) -> Option<Rc<LazyDiscriminants<'a>>> {
```

Comment:
Are we duplicating the discriminants logic here?

This is at quite some risk of drifting in a way that would be really painful to discover: if the "regular" discriminants computation changes but this isn't updated, then our returned discriminants would only be incorrect when the index is applied to queries.

I've had to debug a couple of those bugs already, and it was extremely painful. I'd really prefer if we can make it impossible for us to have another one.

Reply:
As far as I can tell, this is the only instance of the discriminants logic. I have moved it to the new optimizations file, since it is the only place it is currently used, but I'm not sure it's the best place for it, code-grouping wise.